x-pack/plugins/apm/server/plugin.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { PluginInitializerContext, Plugin, CoreSetup } from 'src/core/server';
+import { Observable, combineLatest, AsyncSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Server } from 'hapi';
+import { once } from 'lodash';
+import { Plugin as APMOSSPlugin } from '../../../../src/plugins/apm_oss/server';
+import { createApmAgentConfigurationIndex } from '../../../legacy/plugins/apm/server/lib/settings/agent_configuration/create_agent_config_index';
+import { createApmApi } from '../../../legacy/plugins/apm/server/routes/create_apm_api';
```

Comment:
nit: it's a bit confusing when you have routes in a legacy plugin, being run from a NP plugin, but injecting dependencies from a legacy plugin, just kinda hard to wrap your head around. Do you still need these files in legacy or could you move create_apm_api and all the routes into the NP plugin?

Reply:
I think this makes more sense to do when the migration is completed. Otherwise legacy will import lots of stuff from np and the other way around. I've added it to the list in https://github.com/elastic/kibana/issues/32894